packages/dev/docs/pages/blog/creating-a-pointer-friendly-submenu-experience.mdx

```diff
@@ -0,0 +1,119 @@
+
+Although the timeout solution is simpler than tracking the pointer’s speed, we found that it resulted in a good user experience, so we proceeded with this approach.
+
+## Fault Tolerance
+
+Since our users are human, we want to build in some fault tolerance, but not so much that we invalidate their intent. Here are some ways we did that:
+* **Widen our range of allowed angles**: We added 15 degrees of tolerance to the top and bottom angles. After testing different values, we found that this created the best experience.
+* **Allow invalid movements**: We require that two consecutive invalid pointer movements be made before closing the submenu. Users who experience [tremors](https://en.wikipedia.org/wiki/Tremor) may involuntarily move their pointer in other directions, so it is important to include this feature.  
+
+## Optimizations
+
+We can introduce a few performance optimizations:
+
+* **Throttle**: Most devices refresh their screens 60 times per second. This means that we don't need to do these measurements more frequently than every 16 ms (1 second / 60 = 16.66 ms). We can also experiment with doing checks even less frequently while still maintaining a good user experience. For instance, lowering the sample rate may provide more accurate predictions for users who experience tremors.
+* **Track movements only when necessary**: We can start tracking pointer movements when the submenu opens and stop tracking when the submenu closes, or the pointer reaches the submenu. 
+* **Calculate angles only when necessary**: If the pointer is moving in the opposite direction of the submenu, there's no need to calculate and compare angles. We can check this by comparing the pointer’s delta x to the submenu’s closest edge's x.
+* **Check the pointer event type**: We can check the pointer event's [pointerType](https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent/pointerType) property and ignore the event if it is type 'pen' or 'touch'.
+
+## Alternatives
+
+Let's compare our approach to a few other methods:
+
+* **Timeout only**: We could use just a timeout instead of incorporating the direction the pointer moves. The downside of this is that moving the pointer vertically between parent menu items would cause delayed interactions that could be unpleasant or unexpected for the user.
+* **Delay before opening submenus**: We could introduce a delay before opening each submenu, but that would introduce a similar negative user experience as the timeout-only method described above.
+* **Check if the point is within a triangle**: We could define a triangle from the previous pointer position to the top inside corner of the submenu and the bottom inside of the submenu. This method is outlined by Ben Kamens in [Breaking down Amazon’s mega dropdown](https://bjk5.com/post/44698559168/breaking-down-amazons-mega-dropdown).
```

Comment:
Can we add something here about how this differs from our approach and why we didn't go with it? It seem very similar.

Reply:
Yeah, based on their wording I assumed they were doing something like a [point-in-polygon](https://en.wikipedia.org/wiki/Point_in_polygon) method, but looks like they are actually using the slope formula. This is similar to our solution, but ours ends up comparing values in radians via [Math.atan2](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/atan2).